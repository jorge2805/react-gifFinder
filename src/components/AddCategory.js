import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {
  
    const [inputValue, setinputValue] = useState("")

    const onChangeInput = (e) =>{
        setinputValue(e.target.value)
    }

    const onSubmitForm = (e) =>{
        e.preventDefault();
        
        if (inputValue.trim().length >= 2) {
            setCategories(categories => [inputValue, ...categories]);
            setinputValue("");
        }        
    }

    return (    
    <>
    <form onSubmit={onSubmitForm}>
        <input
            type="text"
            value={inputValue}
            onChange = {onChangeInput}
        />
    </form>
    </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}