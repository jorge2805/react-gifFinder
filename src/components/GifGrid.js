import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifts } from '../helpers/getGifts';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';



export const GifGrid = ({category}) => {

    const {data,loading} = useFetchGifs(category);

    return (
    <>
    <h3 className='animate__animated animate__bounce'>{category}</h3>
    <p className='animate__animated animate__flash'>{loading ? "Cargando..." : null}</p>
    <div className='card-grid'>       
        {
            data.map( image =>                
                <GifGridItem 
                key={image.id}
                {...image}                
                />
            )                
        }
    </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
