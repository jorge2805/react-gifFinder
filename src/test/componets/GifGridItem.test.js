import React from 'React'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en GifGridItem', () => { 
    
    const img ={
        id: 1,
        title: "img.title",
        url: "img.images.downsized_medium.url"
    }

    const wrapper = shallow( <GifGridItem {...img} /> )

    test('Probando snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de existir un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(img.title);
    });

    test('Evaluando Propiedades Imagen', () => {
        const img2 = wrapper.find('img').props();

        expect(img2.src).toBe(img.url);
        expect(img2.alt).toBe(img.title);

    })

    test('debe tener fade_in', () => {
        const Div = wrapper.find('div').prop('className');
        expect(Div.includes("animate__bounce")).toBe(true);

    })

})