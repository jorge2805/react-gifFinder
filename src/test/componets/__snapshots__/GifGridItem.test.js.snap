// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Probando snapshot 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="img.title"
    src="img.images.downsized_medium.url"
  />
  <p>
    img.title
  </p>
</div>
`;
